### PR TITLE
support rolling on jammy

### PR DESCRIPTION
**Pull request by wep21**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (add tickets here #1) |
| Primary OS tested on | (Ubuntu22.04, build) |
| Robotic platform tested on | (Steve's Robot, gazebo simulation of Tally, hardware turtlebot) |

---

## Description of contribution in a few bullet points

This PR enables build on rolling distribution on Ubuntu22.04.
* Update tbb api(See https://www.intel.com/content/dam/develop/external/us/en/documents/tbbrevamp.pdf)
* Use system cmake macro to find tbb
* Update rosidl cmake api(See https://docs.ros.org/en/rolling/Releases/Release-Humble-Hawksbill.html)
* Suppress cmake warning(See https://cmake.org/cmake/help/v3.17/module/FindPackageHandleStandardArgs.html)

## Description of documentation updates required from your changes

---

### CODE REVIEW

**Comment by wep21** [comment]
cc @SteveMacenski Could you review this PR?

File: lib/karto_sdk/CMakeLists.txt [hunk 1]
```diff
@@ -29,7 +32,12 @@ add_definitions(${EIGEN3_DEFINITIONS})
 include_directories(include ${EIGEN3_INCLUDE_DIRS} ${TBB_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
 add_library(kartoSlamToolbox SHARED src/Karto.cpp src/Mapper.cpp)
 ament_target_dependencies(kartoSlamToolbox ${dependencies})
-target_link_libraries(kartoSlamToolbox ${Boost_LIBRARIES} ${TBB_LIBRARIES})
+if(${TBB_VERSION} VERSION_GREATER_EQUAL 2021.1)
+  target_link_libraries(kartoSlamToolbox ${Boost_LIBRARIES} TBB::tbb)
+else()
+  target_link_libraries(kartoSlamToolbox ${Boost_LIBRARIES} ${TBB_LIBRARIES})
+  add_compile_definitions(USE_DEPRECATED_TBB_API)
```
Comment (SteveMacenski) [inline lib/karto_sdk/CMakeLists.txt]: Is `USE_DEPRECATED_TBB_API` for supporting 22.04's change of TBB? If so, you can just use the new APIs and don't need to support backwards. There are other branches for galactic/foxy/etc, this is for bleeding edge `rolling` so we can make API breaking changes here safely. Unless the old version is used in 22.04/Humble, we don't need to support it

File: CMakeLists.txt [hunk 2]
```diff
@@ -141,7 +153,12 @@ ament_target_dependencies(toolbox_common
   ${dependencies}
 )
 target_link_libraries(toolbox_common kartoSlamToolbox ${Boost_LIBRARIES})
-rosidl_target_interfaces(toolbox_common ${PROJECT_NAME} "rosidl_typesupport_cpp")
+if(${rosidl_cmake_VERSION} VERSION_LESS 2.5.0)
+  rosidl_target_interfaces(toolbox_common ${PROJECT_NAME} "rosidl_typesupport_cpp")
```
Comment (SteveMacenski) [inline CMakeLists.txt]: As well here, are these changes required for Humble/22.04? If so, we don't need to support backwards, there are other branches for that

File: CMakeLists.txt [hunk 3]
```diff
@@ -132,7 +139,12 @@ target_link_libraries(ceres_solver_plugin ${CERES_LIBRARIES}
                                           ${TBB_LIBRARIES}
                                           kartoSlamToolbox
 )
-rosidl_target_interfaces(ceres_solver_plugin ${PROJECT_NAME} "rosidl_typesupport_cpp")
+if(${rosidl_cmake_VERSION} VERSION_LESS 2.5.0)
+  rosidl_target_interfaces(ceres_solver_plugin ${PROJECT_NAME} "rosidl_typesupport_cpp")
```
Comment (SteveMacenski) [inline CMakeLists.txt]: Ditto

File: CMakeLists.txt [hunk 4]
```diff
@@ -119,7 +121,12 @@ target_include_directories(SlamToolboxPlugin PUBLIC
   ${OGRE_INCLUDE_DIRS}
 )
 target_link_libraries(SlamToolboxPlugin ${QT_LIBRARIES} rviz_common::rviz_common)
-rosidl_target_interfaces(SlamToolboxPlugin ${PROJECT_NAME} "rosidl_typesupport_cpp")
+if(${rosidl_cmake_VERSION} VERSION_LESS 2.5.0)
+  rosidl_target_interfaces(SlamToolboxPlugin ${PROJECT_NAME} "rosidl_typesupport_cpp")
```
Comment (SteveMacenski) [inline CMakeLists.txt]: Ditto

File: CMakeLists.txt [hunk 5]
```diff
@@ -72,9 +72,11 @@ set(libraries
 
 find_package(PkgConfig REQUIRED)
 find_package(Eigen3 REQUIRED)
+set(FPHSA_NAME_MISMATCHED 1) # Suppress warnings, see https://cmake.org/cmake/help/v3.17/module/FindPackageHandleStandardArgs.html
```
Comment (SteveMacenski) [inline CMakeLists.txt]: Should we just update names?

**Review by SteveMacenski — COMMENTED** [review]
Sorry it took a bit, I was on vacation. I really appreciate the help to get this to 22.04 and the thoroughness. I think though we can just drop outdated support from this branch!

File: lib/karto_sdk/CMakeLists.txt (hunk 1 above)
Comment (wep21) [reply lib/karto_sdk/CMakeLists.txt]: Yes. I missed the old distribution branches, so I removed backward support at https://github.com/SteveMacenski/slam_toolbox/pull/494/commits/ec110d31cdd0365afb9c2d281eb68d16fa20b9dc.

File: CMakeLists.txt (hunk 2 above)
Comment (wep21) [reply CMakeLists.txt]: Yes, these changes are for humble/rolling api change. I also removed them.

File: CMakeLists.txt (hunk 5 above)
Comment (wep21) [reply CMakeLists.txt]: I changed upper and lower case letters for SuiteSparse library at https://github.com/SteveMacenski/slam_toolbox/pull/494/commits/76cba46c03f3b88534e1ee0ac05627b7ddb41654.

**Review by SteveMacenski — APPROVED** [review]
LGTM
